tools/stress-cluster/services/Stress.Watcher/src/PodEventHandler.cs

```diff
@@ -215,5 +228,61 @@ public bool ShouldStartPodChaos(WatchEventType type, V1Pod pod)
 
             return true;
         }
+
+        public async Task DeleteResources(V1Pod pod)
```

Comment:
Should this clean-up happen via a call to https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/Remove-TestResources.ps1 instead?

Reply:
Yes the duplicated logic is my main concern. However for the most part the remove just deletes resource groups. You are correct that the purging logic may also need to be duplicated if we find there are expensive resources to be purged.

However it is most important that we keep the tagging the same so that our scheduled clean-up jobs can still clean these resources up as well.

Reply:
Yep, tagging will work the same because the deployment uses the New-TestResources script as part of an init container and not the stress watcher. We'll have to keep an eye out for persistent failures (e.g. watcher always failing and us not noticing because of the pipeline cleanup job), but otherwise the tags should conform.